catalogue_graph/src/ingestor/extractors/base_extractor.py

```diff
@@ -92,20 +89,16 @@ def make_neptune_query(
         if query_type in EXPENSIVE_QUERIES:
             chunk_size = NEPTUNE_EXPENSIVE_CHUNK_SIZE
 
-        def _run_query(chunk: Iterable[str]) -> list[dict]:
-            return self.neptune_client.run_open_cypher_query(
-                NEPTUNE_QUERIES[query_type],
-                self.neptune_params | {"ids": sorted(chunk)},
-            )
-
         start = time.time()
-        raw_results = process_stream_in_parallel(
-            ids, _run_query, chunk_size, QUERY_THREAD_COUNT
+        results = self.neptune_client.run_parallel_query(
```

Comment:
I like that logic from `process_stream_in_parallel` has been pushed into `neptune_client` as `run_parallel_query`.